### PR TITLE
[ISSUE #2657] Method uses immediate execution of a block of code that is often not used [RemoteSubscribeEventProcessor] 

### LINKED ISSUE #2657: [Enhancement] Method uses immediate execution of a block of code that is often not used [RemoteSubscribeEventProcessor]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
image
located at:
eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java line 131,151
analysis and explanation:
This method uses the Optional.orElse() method passing in some code that will execute immediately, whether or not the else case of the Optional is needed. This may cause incorrect side effects to happen, or at the minimum, code to execute for no reason.

### Describe the solution you'd like

It would be better to use Optional.orElseGet()

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by chenyi19851209**
Fixes #2657.

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java [hunk 1]
```diff
@@ -104,11 +103,15 @@ public void handler(HandlerService.HandlerSpecific handlerSpecific, HttpRequest
 
         //validate body
         byte[] requestBody = requestWrapper.getBody();
-
-        Map<String, Object> requestBodyMap = Optional.ofNullable(JsonUtils.deserialize(
-            new String(requestBody, Constants.DEFAULT_CHARSET),
-            new TypeReference<HashMap<String, Object>>() {}
-        )).orElseGet(Maps::newHashMap);
+        Map<String, Object> requestBodyMap;
```
Comment (jonyangx) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java:106]: The code is redundant than before. @chenyi19851209

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java [hunk 2]
```diff
@@ -104,11 +103,15 @@ public void handler(HandlerService.HandlerSpecific handlerSpecific, HttpRequest
...
+        if(requestBody != null){
```
Comment (jonyangx) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java:107]: The code is redundant than before.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java [hunk 3]
```diff
@@ -104,11 +103,15 @@ public void handler(HandlerService.HandlerSpecific handlerSpecific, HttpRequest
...
+            requestBodyMap = JsonUtils.deserialize(
```
Comment (jonyangx) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java:108]: The code is redundant than before. @chenyi19851209

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java [hunk 4]
```diff
@@ -122,10 +125,15 @@ public void handler(HandlerService.HandlerSpecific handlerSpecific, HttpRequest
 
 
         // SubscriptionItem
-        List<SubscriptionItem> subscriptionList = Optional.ofNullable(JsonUtils.deserialize(
-            topic,
-            new TypeReference<List<SubscriptionItem>>() {}
-        )).orElseGet(Collections::emptyList);
+        List<SubscriptionItem> subscriptionList;
```
Comment (jonyangx) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java:128]: The code is redundant than before. @chenyi19851209

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java [hunk 5]
```diff
@@ -200,10 +208,15 @@ public void handler(HandlerService.HandlerSpecific handlerSpecific, HttpRequest
             String remoteResult = post(closeableHttpClient, targetMesh, builderRemoteHeaderMap(localAddress), remoteBodyMap,
                 response -> EntityUtils.toString(response.getEntity(), Constants.DEFAULT_CHARSET));
 
-            Map<String, String> remoteResultMap = Optional.ofNullable(JsonUtils.deserialize(
-                remoteResult,
-                new TypeReference<Map<String, String>>() {}
-            )).orElseGet(Maps::newHashMap);
+            Map<String, String> remoteResultMap;
```
Comment (jonyangx) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/RemoteSubscribeEventProcessor.java:211]: The code is redundant than before. @chenyi19851209

**Comment by xwm1992** [comment]
@chenyi19851209 Since this issue has been fixed, I'll close this pr.

**Pull request closed without merging**